### ISSUE TITLE
Use input_ids as labels instead of 1s when training

### ISSUE THREAD

**Issue by kabachuha** [oobabooga/text-generation-webui#4021]
**Description**

Hi, all!

Long story short, I stumbled upon a very strange thing when looking through the training code: while masking the undesired tokens with -100s, it also marks *all the labels*, which are the tokens what the model is supposed to fit, with 1s. I looked up Transformers Trainer docs and the source code and I didn't found the evidence that it defaults to input_ids when the labels are 1s. Am I missing something? The fun thing is that the model still trains from the input context, so while it may be worth to preserve as a legacy mode, I'd be nice to have a switch for a more adequate behavior.

**Additional Context**

See Alpaca lora's code for their training implementation

https://github.com/tloen/alpaca-lora/blob/8bb8579e403dc78e37fe81ffbb253c413007323f/finetune.py#L144

**Comment by kabachuha** [comment]
Oh, I see you're using the Data Collator https://huggingface.co/docs/transformers/main_classes/data_collator, which should replace unmasked values with targets.

All fine then
